#! /usr/bin/env bash

# This script is meant to generate a simulated read dataset of known referene genomes using ART.
# The public dataset we are using is Illumina paired-end metagenomic reads generated using the HiSeq 4000 (2 x 150bp).

# The "art_illumina" command is used to simulate paired-end reads off of an Illumina sequencing platform using the following parameters:
    # 1. "-ss": HiSeq 2500 sequencing platform to simulate reads generated by HiSeq 4000 since that option was unavailable (both had 2 x 150bp read lengths)
    # 2. "-sam": A SAM alignment file is generated for each reference organism
    # 3. "-i": Input reference FASTA file of select whole genomes
    # 4. "-p": Indicates paired-end sequencing simulation mode
    # 5. "-l": Length of each read
    # 6. "-c": Number of each read pair produced
    # 7. "m": Mean length of each read
    # 8. "-s": Standard deviation of each read
    # 9. "-o": Output file name
    # 10. "-M": CIGAR method to introduce mismatches instead of default method

art_illumina -ss HS25 -sam -i Escherichia_coli.fasta -p -l 150 -c 20 -m 200 -s 10 -o Escherichia_coli_paired -M

art_illumina -ss HS25 -sam -i Bacillus_anthracis.fasta -p -l 150 -c 200 -m 200 -s 10 -o Bacillus_anthracis_paired -M

art_illumina -ss HS25 -sam -i Staphylococcus_aureus.fasta -p -l 150 -c 1000 -m 200 -s 10 -o Staphylococcus_aureus_paired -M

art_illumina -ss HS25 -sam -i Enterococcus_faecalis.fasta -p -l 150 -c 500 -m 200 -s 10 -o Enterococcus_faecalis_paired -M

art_illumina -ss HS25 -sam -i Streptococcus_agalactiae.fasta -p -l 150 -c 850 -m 200 -s 10 -o Streptococcus_agalactiae_paired -M
